lifetiles-graph/src/main/java/nl/tudelft/lifetiles/annotation/controller/BookmarkController.java

```diff
@@ -0,0 +1,181 @@
+package nl.tudelft.lifetiles.annotation.controller;
+
+import java.net.URL;
+import java.util.List;
+import java.util.ResourceBundle;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.collections.transformation.FilteredList;
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.VBox;
+import javafx.util.Callback;
+import nl.tudelft.lifetiles.annotation.model.AbstractBookmark;
+import nl.tudelft.lifetiles.annotation.model.GeneAnnotation;
+import nl.tudelft.lifetiles.annotation.model.KnownMutation;
+import nl.tudelft.lifetiles.core.controller.AbstractController;
+import nl.tudelft.lifetiles.core.util.Message;
+
+/**
+ * Controller for the Bookmark sidebar.
+ * @author Albert Smit
+ *
+ */
+public class BookmarkController extends AbstractController {
+
+    /**
+     * The wrapper element for the control.
+     */
+    @FXML
+    private VBox wrapper;
+
+    /**
+     * A listView to show all annotations.
+     */
+    @FXML
+    private ListView<AbstractBookmark> bookmarkList;
+
+    /**
+     * the searchbox to search through all annotations.
+     */
+    @FXML
+    private TextField searchBox;
+
+    /**
+     * The list of all bookmarks.
+     */
+    private ObservableList<AbstractBookmark> listItems;
+    /**
+     * The filtered list that will wrap the listItems.
+     * do not add items directly to this list, instead add them to listItems.
+     */
+    private FilteredList<AbstractBookmark> filteredItems;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public final void initialize(final URL location,
+            final ResourceBundle resources) {
+        hide();
+
+        // wrap the observable list in a filtered list so we can search it later
+        listItems = FXCollections.observableArrayList();
+        filteredItems = new FilteredList<AbstractBookmark>(listItems);
+        bookmarkList.setItems(filteredItems);
+
+        /*
+         * create a new Cell factory so we can use a different method
+         * to create the text of each cell,
+         * and assign an event handler for mouse clicks.
+         */
+        bookmarkList.setCellFactory(new Callback<ListView<AbstractBookmark>,
```

Comment:
the inner anonymous class for `Callback` can be replaced by a lambda to make the code more readable and less bulky.